### PR TITLE
kv: less error tolerance in txn correctness tests

### PR DESCRIPTION
**Pull request by tbg**
Removed the nondescript "expectError" boolean in the txn correctness tests. There
was only one test which relied on it (which was easily refactored).

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/6971)

### CODE REVIEW

**Comment by bdarnell** [comment]
:lgtm:

<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> #### kv: less error tolerance in txn correctness tests
> 
> Removed the nondescript "expectError" boolean in the txn correctness tests. There
> 
> was only one test which relied on it (which was easily refactored).
> 
> 
> 
</details>


---

Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[kv/txn_correctness_test.go, line 937 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6971#-KJ6tBHiLzPgoICVDaYE:-KJ6tBHiLzPgoICVDaYF:-1876095021) ([raw file](https://github.com/cockroachdb/cockroach/blob/07eed87d780aef606bebe7bc3aecc9a18b28de0f/kv/txn_correctness_test.go#L937)):*
> ```Go
> // 3). Snapshot isolation, however, may not notice any conflict (see
> // history above) and may set A=1, B=1.
> func TestTxnDBWriteSkewAnomaly_serializable(t *testing.T) {
> ```

Rename this back?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6971#-:-KJ6tJUIP_R2dXdu6vsx:1417088721)*

**Comment by tschottdorf** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[kv/txn_correctness_test.go, line 937 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6971#-KJ6tBHiLzPgoICVDaYE:-KJ6xJZCWXUPfWVrKpXI:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/07eed87d780aef606bebe7bc3aecc9a18b28de0f/kv/txn_correctness_test.go#L937)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary>
> Rename this back?
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6971)*

File: kv/txn_correctness_test.go [hunk 1]
```diff
@@ -911,17 +936,6 @@ func TestTxnDBPhantomDeleteAnomaly(t *testing.T) {
...
-		checkFn: func(env map[string]int64) error {
-			if !((env["A"] == 1 && env["B"] == 2) || (env["A"] == 2 && env["B"] == 1)) {
-				return util.Errorf("expected either A=1, B=2 -or- A=2, B=1, but have A=%d, B=%d", env["A"], env["B"])
-			}
-			return nil
-		},
-	}
-	checkConcurrency("write skew", onlySerializable, []string{txn1, txn2}, verify, true, t)
-	checkConcurrency("write skew", onlySnapshot, []string{txn1, txn2}, verify, false, t)
+	runWriteSkewTest(t, roachpb.SERIALIZABLE)
```
Comment (spencerkimball) [inline kv/txn_correctness_test.go]: I think using the helper method is less understandable than switching back to the old code in this method and just putting a switch statement for the iso type in the `checkFn`.

**Comment by spencerkimball** [comment]
I don't see what changed here which would fix the intermittent test failures we've been seeing...?

**Comment by tschottdorf** [comment]
I'm not claiming that it fixes them.

On Tue, May 31, 2016 at 3:25 PM Spencer Kimball <notifications@github.com>
wrote:

> I don't see what changed here which would fix the intermittent test
> failures we've been seeing...?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/6971#issuecomment-222793422>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AE135GnQCm3jHHodRtm8cWsMhFgE4CQyks5qHIsUgaJpZM4IqqSW>
> .
>
-- 

-- Tobias

File: kv/txn_correctness_test.go (hunk 1 above)
Comment (tschottdorf) [inline kv/txn_correctness_test.go]: I don't have the isolation level available in checkFn though, and I doubt that plumbing it there just for that is a better solution.

**Comment by spencerkimball** [comment]
Oh, this is just an related cleanup?

**Comment by tschottdorf** [comment]
Yep

**Pull request merged**